lib/datadog/core/configuration/option.rb

```diff
@@ -102,6 +191,33 @@ def context_eval(&block)
           @context.instance_eval(&block)
         end
 
+        def set_value_from_env_or_default
+          value = nil
+          precedence = nil
+
+          if definition.env && ENV[definition.env]
+            value = coerce_env_variable(ENV[definition.env])
+            precedence = Precedence::PROGRAMMATIC
```

Comment:
I don't dislike the idea. 

For now, I would not do much about it because there is another PR by @marcotc https://github.com/DataDog/dd-trace-rb/pull/2972 that uses the precedence to unset configuration values.

I'm not sure what weight `Precedence::ENVIRONMENT` should have. Same as `Precedence::PROGRAMMATIC`? 

What would happen when someone configures something via ENV var and later changed it using the `configure` block? 

Since those are question that we might not have the answer now I think is best to wait until making that change